pkg/network/namescheme/networknamescheme.go

```diff
@@ -20,37 +20,44 @@
 package namescheme
 
 import (
+	"crypto/sha256"
 	"fmt"
+	"io"
 
 	v1 "kubevirt.io/api/core/v1"
 
 	"kubevirt.io/kubevirt/pkg/network/vmispec"
 )
 
-const PrimaryPodInterfaceName = "eth0"
+const (
+	// MaxIfaceNameLen equals max kernel interface name len (15) - length("-nic")
+	// which is the suffix used for the bridge binding interface with IPAM.
+	// (the interface created to hold the pod's IP address - and thus appease CNI).
+	MaxIfaceNameLen         = 11
+	PrimaryPodInterfaceName = "eth0"
+)
 
 // CreateNetworkNameScheme iterates over the VMI's Networks, and creates for each a pod interface name.
 // The returned map associates between the network name and the generated pod interface name.
-// Primary network will use "eth0" and the secondary ones will use "net<id>" format, where id is an enumeration
-// from 1 to n.
+// Primary network will use "eth0" and the secondary ones will be named "net<id>" where id is the network name sha256.
 func CreateNetworkNameScheme(vmiNetworks []v1.Network) map[string]string {
 	networkNameSchemeMap := mapMultusNonDefaultNetworksToPodInterfaceName(vmiNetworks)
-
 	if multusDefaultNetwork := vmispec.LookUpDefaultNetwork(vmiNetworks); multusDefaultNetwork != nil {
 		networkNameSchemeMap[multusDefaultNetwork.Name] = PrimaryPodInterfaceName
 	}
-
 	return networkNameSchemeMap
 }
 
 func mapMultusNonDefaultNetworksToPodInterfaceName(networks []v1.Network) map[string]string {
 	networkNameSchemeMap := map[string]string{}
-	for i, network := range vmispec.FilterMultusNonDefaultNetworks(networks) {
-		networkNameSchemeMap[network.Name] = secondaryInterfaceName(i + 1)
+	for _, network := range vmispec.FilterMultusNonDefaultNetworks(networks) {
+		networkNameSchemeMap[network.Name] = hashNetworkName(network.Name)
 	}
 	return networkNameSchemeMap
 }
 
-func secondaryInterfaceName(idx int) string {
-	return fmt.Sprintf("net%d", idx)
+func hashNetworkName(networkName string) string {
+	hash := sha256.New()
+	_, _ = io.WriteString(hash, networkName)
+	return fmt.Sprintf("net%x", hash.Sum(nil))[:MaxIfaceNameLen]
```

Comment:
suggestion: `"net-%x"`
hex16 contains letters that make the overall name less readable - a simple dash will fix that.

Reply:
We currently are limiting the length of the interface names to 11 chars (3 of which are the prefix - `net`). 

That leaves 8 characters for the unique part of the name; adding the dash to the prefix will leave only 7 chars for the unique part of the name.

Not saying it is too little, but I also don't find `net-6446d58` more readable than `net6446d58d` (or `net-a446d58` preferable to `neta446d58d`).

Reply:
This is only made worse by the in-pod bridge naming scheme: `k6t-net7e00`. For that one, we have 4 chars available; if we follow your suggestion, we'll have only 3.

Reply:
Maybe it is worth mentioning that humans are not suppose to read this at all.
I do not even understand why we need the `net` part, but I do not really mind.

Reply:
That's a good point. We can live without the `net` prefix as well.

Reply:
i.e. the interfaces would look like (in the pod):
- pod iface: 7e0055a6d51
- bridge name: k6t-7e0055a
- tap device name:  tap7e0055a6

Would this be OK ?